### PR TITLE
Node

### PR DESCRIPTION
**Pull request by webdev03**

### CODE REVIEW

**Comment by hello-smile6** [comment]
Nice!

**Review by hello-smile6 — CHANGES_REQUESTED** [review]
Could you add support for running in headless mode? (For example, on Replit) ?

**Comment by webdev03** [comment]
What's headless mode?

**Comment by webdev03** [comment]
OHH, you mean, with a config thing. I'll set up an authentication thing soon.

**Comment by hello-smile6** [comment]
> OHH, you mean, with a config thing. I'll set up an authentication thing soon.

Could you add support for a configuration project that trusted users could comment on in order to change settings? Also, you might want to find a NodeJS canvas module or something and just automatically delete posts that have UTF-16 whitespace characters, but not delete posts from approved users.

**Review by hello-smile6 — CHANGES_REQUESTED** [review]
JS for loop syntax isn't like Python for loop syntax. To do a for loop on an array:
```
array.forEach(function(itemName) {
  chopUpAndCook(itemName);
};
```
Thought I'd point that out.

**Comment by webdev03** [comment]
Tried that earlier. Doesn't work. I fixed the bug, run the script with Node 16

**Comment by webdev03** [comment]
https://scratch.mit.edu/users/griffpatch/#comments-181316728

**Comment by hello-smile6** [comment]
> Tried that earlier. Doesn't work. I fixed the bug, run the script with Node 16

What was the error? Also, I fixed a syntax error in my code.

**Comment by hello-smile6** [comment]
Sorry if it's a bit disturbing that I went with `chopUpAndCook()` for my example, it's the first thing I thought of.

File: index.js [hunk 1]
```diff
@@ -24,12 +24,12 @@ const print = (query) => console.log(query);
...
   const dom = new JSDOM(commentHTML);
-  const items = dom.window.document.getElementsByClassName("top-level-reply")
+  const items = dom.window.document.getElementsByClassName("top-level-reply");
   for(let elID in items) {
     const element = items[elID];
     if (typeof element == "function") break;
     const commentPoster = element.getElementsByClassName("comment")[0].getElementsByTagName("a")[0].getAttribute('data-comment-user')
-    print(`Current Comment Poster: ${commentPoster}`)
   };
+  print(`Found ${items.length} comments!`)
```
Comment (hello-smile6) [inline index.js:33]: Use `console.log` or define a passthrough function globally.

**Review by hello-smile6 — CHANGES_REQUESTED** [review]
Either create a `print()` function globally or use `console.log()`. `print()` doesn't exist with Node.

File: index.js (hunk 1 above)
Comment (webdev03) [reply index.js:33]: What do you mean?

Comment (webdev03) [reply index.js:33]: I'm replying this through VS Code. So easy to code at the same time :smile:

Comment (hello-smile6) [reply index.js:33]: > I'm replying this through VS Code. So easy to code at the same time 😄

How?

Comment (webdev03) [reply index.js:33]: Github extension

Comment (hello-smile6) [reply index.js:45]: > Github extension

Oh! Do you use the live share extension?

Comment (webdev03) [reply index.js:45]: I have it, but I use it very rarely.

Comment (hello-smile6) [reply index.js:45]: > I have it, but I use it very rarely.

What's it like? If it works on the [web version](https://vscode.dev/) I might use it to sync directories between computers.

**Comment by webdev03** [comment]
Thanks @hello-smile6 for recommending Node, made it 200000000000x easier :)

**Comment by hello-smile6** [comment]
> Thanks @hello-smile6 for recommending Node, made it 200000000000x easier :)

Could I test first?

**Comment by webdev03** [comment]
Sure. Right now it doesn't do any filtering, but if you log the `comments` object after the loop it works.

**Comment by hello-smile6** [comment]
> Sure. Right now it doesn't do any filtering, but if you log the `comments` object after the loop it works.

Okay

**Comment by hello-smile6** [comment]
Hi from vscode! Sharing screenshot on web

**Comment by hello-smile6** [comment]
image

**Comment by hello-smile6** [comment]
@webdev03 Could you look at #6 ?

File: index.js (hunk 1 above)
Comment (hello-smile6) [reply index.js]: > Github extension

I like it! Thanks for the tip.

**Comment by hello-smile6** [comment]
@webdev03 Try this extension for vscode.
> Name: GitHub Notifications
Id: fabiospampinato.vscode-github-notifications-bell
Description: A secure, customizable, statusbar icon that notifies you about notifications on GitHub.
Version: 2.3.1
Publisher: Fabio Spampinato
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-github-notifications-bell

It's nice.

**Review by hello-smile6 — COMMENTED** [review]
My change looks good, just getting gh to acknowledge that I've seen this

**Comment by webdev03** [comment]
I am merging this once I rewrite docs. Should I make a github action which updates NPM, PNPM, and YARN lockfiles?

**Comment by hello-smile6** [comment]
> I am merging this once I rewrite docs. Should I make a github action which updates NPM, PNPM, and YARN lockfiles?

Huh, why not just use NPM? It's bundled with Node.

**Pull request merged**